pandas/tests/reshape/test_merge.py

```diff
@@ -1546,6 +1546,55 @@ def test_dtype_on_categorical_dates(self):
         result_inner = pd.merge(df, df2, how='inner', on=['date'])
         assert_frame_equal(result_inner, expected_inner)
 
```

Comment:
can you parametrize this and make it a single test (with 3 cases)

Reply:
done